### PR TITLE
Faster loading of dashboard and vms/index page

### PR DESCRIPTION
**Pull request by BraunTom**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -122,6 +122,12 @@ group :development do
   gem 'binding_of_caller'
   # Continous execution of unit tests
   gem 'guard' # https://github.com/guard/guard
+  # Benchmarking
+  # Supporting gem for Rails Panel (Google Chrome extension for Rails development)
+  gem 'meta_request' # https://github.com/dejan/rails_panel
+  # code profiler for MRI Ruby
+  gem 'ruby-prof' # https://github.com/ruby-prof/ruby-prof
```
Comment (chrisma) [inline Gemfile]: Should `ruby-prof-rails` be used instead? https://github.com/tleish/ruby-prof-rails

File: app/api/v_sphere/folder.rb [hunk 2]
```diff
@@ -83,7 +90,11 @@ def create_vm(cpu, ram, capacity, name, cluster)
 
       vm_config = creation_config(cpu, ram, capacity, add_prefix(name), cluster.networks.first)
       vm = @folder.CreateVM_Task(config: vm_config, pool: cluster.resource_pool).wait_for_completion
-      VSphere::VirtualMachine.new vm
+      VSphere::VirtualMachine.new vm, folder: self
+    end
+
+    def equal?(other)
```
Comment (chrisma) [inline app/api/v_sphere/folder.rb]: `equal?` has a specific purpose in Ruby, i.e. to

> determine object identity (that is, a.equal?(b) if and only if a is the same object as b)

(http://ruby-doc.org/core-2.6.2/Object.html#method-i-eql-3F)

Therefore "Unlike ==, the equal? method should never be overridden by subclasses"

Most likely, `eql?` would be a better fit here

File: Gemfile (hunk 1 above)
Comment (BraunTom) [reply Gemfile]: I tried to install it but got an error and Ruby-prof is good enough

File: app/api/v_sphere/folder.rb (hunk 2 above)
Comment (BraunTom) [reply app/api/v_sphere/folder.rb]: changed it to eql?

File: Gemfile [hunk 3]
```diff
@@ -122,6 +122,12 @@ group :development do
   gem 'binding_of_caller'
   # Continous execution of unit tests
   gem 'guard' # https://github.com/guard/guard
+  # Benchmarking
+  # Supporting gem for Rails Panel (Google Chrome extension for Rails development)
+  gem 'meta_request' # https://github.com/dejan/rails_panel
```
Comment (BjoernDaase) [inline Gemfile]: Do we really want to merge the `meta-request` and `ruby-prof` gems into `dev` or should we remove them before merging?

File: app/api/v_sphere/virtual_machine.rb [hunk 4]
```diff
@@ -339,7 +352,7 @@ def request
       @request ||= Request.accepted.find_by name: name
     end
 
-    private
+    #private
```
Comment (BjoernDaase) [inline app/api/v_sphere/virtual_machine.rb]: Why is the `private` commented? Where do we need them to be non-private?

File: db/schema.rb [hunk 5]
```diff
@@ -30,9 +30,9 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.string "vsphere_root_folder", default: ""
-    t.string "puppet_init_path", default: "/"
-    t.string "puppet_classes_path", default: "/Name"
-    t.string "puppet_nodes_path", default: "/Node"
+    t.string "puppet_init_path"
```
Comment (BjoernDaase) [inline db/schema.rb]: Due to @chrisma we should not have db changes in MRs.

File: Gemfile (hunk 3 above)
Comment (BraunTom) [reply Gemfile]: The loading time is better but not optimal so when we need to benchmark later we have the gems already installed

File: app/api/v_sphere/folder.rb [hunk 6]
```diff
@@ -11,17 +11,24 @@ class Folder
...
+      @parent = parent
+      @folder_name = name
     end
 
-    def parent
-      @folder.parent
+    def parent(lookup: true)
+      if @parent.nil? && lookup
+        parent = @folder.parent
+        parent.nil? ? nil : VSphere::Folder.new(parent)
```
Comment (MrModder) [inline app/api/v_sphere/folder.rb]: We should cache the parent here, e.g.
``` ruby
@parent = parent.nil? nil : VSphere::Folder.new parent
```

File: app/api/v_sphere/virtual_machine.rb [hunk 7]
```diff
@@ -88,16 +88,21 @@ def self.user_vms(user)
...
-      if /^\d{8}_vm-/.match? @vm.name
-        @vm.name[12..-1]
+      return @name unless @name.nil?
+
+      @vsphere_name ||= @vm.name
+      if /^\d{8}_vm-/.match? @vsphere_name
+        @vsphere_name[12..-1]
       else
-        @vm.name
+        @vsphere_name
```
Comment (MrModder) [inline app/api/v_sphere/virtual_machine.rb]: Again, update the cache here (@name)

Comment (MrModder) [reply app/api/v_sphere/virtual_machine.rb]: And two lines above

File: app/api/v_sphere/virtual_machine.rb [hunk 8]
```diff
@@ -147,11 +156,11 @@ def reboot_guest_os
     # We do not provide a power_state? method which just returns a boolean, because vSphere can internally handle
     # more than just two power states and we might later need to respond to more states than just two
     def powered_on?
-      @vm.summary.runtime.powerState == 'poweredOn'
+      summary.runtime.powerState == 'poweredOn'
```
Comment (MrModder) [inline app/api/v_sphere/virtual_machine.rb]: 👍

File: app/helpers/vms_helper.rb [hunk 9]
```diff
@@ -31,6 +31,15 @@ def status_for(vm)
     end
   end
 
+  def status_color(vm)
```
Comment (MrModder) [inline app/helpers/vms_helper.rb]: 👍

**Pull request merged**